RC2--Webforms, Options, Metadata Display Templates, Configurations

alliomeria opened the pull request:
For dropping off RC2 Webforms, Webform Options, Metadata Display Templates, other related configurations.

@DiegoPino, please let me know if I can prepare any of these files differently to be more readily helpful/useful for RC2. Please also let me know if any of these need additional corrections or revisions. Thank you very much!

DiegoPino commented:
@alliomeria wow. This is just wonderful, so much good work! About your questions
1.  Webforms & Related: We need same UUIDs as in archipelago-deployment, or RC1 to RC2 update people will end with duplicated entries but same Label (and humans may get confused). This is also a future conversation, maybe we could after 1.0.0 always ship with new UUIDS? That way we never overwrite their old (customized) ones? but that also means changing sync completely to match the new ones for every release. Maybe a future conversation, so for now, let's reuse the UUIDs.
2. Metadata Display Templates:
All extensions need to be twig.html (to be consistent in the future) and we may want to maintain the ID based name order to make automatic ingest/update easier (and the UUIDs later when I transform them into JSON API JSON representation for ingest like here: https://github.com/esmero/archipelago-deployment/blob/1.0.0-RC2/d8content/metadatadisplay_entity_01.json.  All your new ones can come after if you are Ok with that and you can name them whatever you want, basically

> If so, would it be helpful if I prepared these to match the existing, add the additional Simple DC and MODS 3.7 as 12 & 13 respectively?
Yes!

3.- Schema.org Creative Works Lists: I would not remove the ones we are not mapping, instead better map them to something  meaningful except for Video Object, not sure what the difference is between Video and Video Object? Do you think we should ship with a very, very basic Panorama Tour webform?

- Digital Document should probably map to the PDF viewer View Mode.
- Data Set. To be determined. Should I have a View Mode setup that has No viewers at all?
- Archive Component: Should we ask people if having Finding AIDs / Template and a simpler Form for that one is desired for 1.0.0?

DiegoPino commented on RC2/format_strawberryfield.viewmodemapping_settings.yml:
```diff
@@ -25,7 +25,7 @@ type_to_viewmode:
     active: true
     weight: -6
   -
-    jsontype: Datataset
```
=)

alliomeria replied:
(=

alliomeria commented:
@DiegoPino great, very glad these are looking pretty good to you! Thank you for your review!

I made a couple tiny changes based off of your feedback, and have a few more questions...

1.) The Webforms and Webforms Options already have the same UUIDs as those found in archipelago-deployment (makes sense :blush:). Regarding how to handle for future releases, I agree that it might helpful to switch to new UUIDs at some point. Would you want to also ship with additional form modes and configurations as well then? Or leave that for users to setup for the new/updated default webform versions only if desired?

2.) I updated the extensions to twig.html for the various Metadata Display Templates. Some of the existing templates here (https://github.com/esmero/archipelago-deployment/tree/1.0.0-RC2/d8content/metadatadisplays) have twig.txt extensions, but I went with twig.html as specified and matched the existing template names if present. I would be happy to help package these as needed for the JSON representation, but I would need a refresher on how to do so (apologies for not being more familiar with this transformation process—I know we reviewed it for RC1 preparations; can’t seem to find the screenshots I took from that particular meeting/review session).

3. Ok, understood about not removing any from the Schema.org Creative Works List. 
- About the Movie/Video/Video Object types, my notes from earlier today were not very clear—sorry about the confusion. VIdeo is not present on the current (RC1) schema.org options list, but is/was being used in the View Mode mappings for RC1 (https://github.com/esmero/archipelago-deployment/blob/1.0.0-RC1/config/sync/format_strawberryfield.viewmodemapping_settings.yml). In the mappings list I added here for potential use in RC2, I have mappings for both Movie and Video Object. Video is a Property in Schema.org (https://schema.org/video). Both Video Object (https://schema.org/VideoObject) and Movie (https://schema.org/Movie) are Types. I think there may a place for both Types, but perhaps one is better than the other?
- The mapping for Digital Document has been changed to the PDF viewer.
- Should I remove the Dataset mapping entirely then? (Just corrected the jsontype typo)
- For Archive Component, yes, I think reaching out to ask people about having a Finding AID template and a corresponding Form is a great idea for 1.0.0! Are you thinking a google group post, or would you like me to open a GitHub Issue about this?

Thanks again for your review and recommendations!

DiegoPino commented:
> @DiegoPino great, very glad these are looking pretty good to you! Thank you for your review!

😸 

> I made a couple tiny changes based off of your feedback, and have a few more questions...

❤️ 

> 1.) The Webforms and Webforms Options already have the same UUIDs as those found in archipelago-deployment (makes sense 😊). Regarding how to handle for future releases, I agree that it might helpful to switch to new UUIDs at some point. Would you want to also ship with additional form modes and configurations as well then? Or leave that for users to setup for the new/updated default webform versions only if desired?

For now let's ship with what we have, same single form Mode and configs, since you are reusing the UUIDs. If we go for a special Panorama Tour Webform we may use that to teach people how to setup their own Form modes. I feel no need to add more configs for this release

> 2.) I updated the extensions to twig.html for the various Metadata Display Templates. Some of the existing templates here (https://github.com/esmero/archipelago-deployment/tree/1.0.0-RC2/d8content/metadatadisplays) have twig.txt extensions, but I went with twig.html as specified and matched the existing template names if present. I would be happy to help package these as needed for the JSON representation, but I would need a refresher on how to do so (apologies for not being more familiar with this transformation process—I know we reviewed it for RC1 preparations; can’t seem to find the screenshots I took from that particular meeting/review session).

Cool. Let's go through this procedure tomorrow. We may automatize the release process (or document it!) via some scripts but it won't hurt if we do it manually first.

> 1. Ok, understood about not removing any from the Schema.org Creative Works List.

Thx

> * About the Movie/Video/Video Object types, my notes from earlier today were not very clear—sorry about the confusion. VIdeo is not present on the current (RC1) schema.org options list, but is/was being used in the View Mode mappings for RC1 (https://github.com/esmero/archipelago-deployment/blob/1.0.0-RC1/config/sync/format_strawberryfield.viewmodemapping_settings.yml). In the mappings list I added here for potential use in RC2, I have mappings for both Movie and Video Object. Video is a Property in Schema.org (https://schema.org/video). Both Video Object (https://schema.org/VideoObject) and Movie (https://schema.org/Movie) are Types. I think there may a place for both Types, but perhaps one is better than the other?

I now remember. Since the View Mode mapper uses "any" `type`:`Value` key found in the JSON, by having Video further down in the list we can catch any ADO that has a Video File (because as:video will contain files that have "type":"Video"). Same goes for Document, Image, etc. So we can keep both, Video Object and Movie Further up so they are selected first, and if not, as last resort, Video. Does that make sense?

> * The mapping for Digital Document has been changed to the PDF viewer.

Great!

> * Should I remove the Dataset mapping entirely then? (Just corrected the jsontype typo)

I think I should take care of that one... Leave it for now

> * For Archive Component, yes, I think reaching out to ask people about having a Finding AID template and a corresponding Form is a great idea for 1.0.0! Are you thinking a google group post, or would you like me to open a GitHub Issue about this?

Google Group is more open I guess. Thanks!


> Thanks again for your review and recommendations!

Thanks to you! Good good work as always!

alliomeria commented:
Hello again @DiegoPino!

> For now let's ship with what we have, same single form Mode and configs, since you are reusing the UUIDs. If we go for a special Panorama Tour Webform we may use that to teach people how to setup their own Form modes. I feel no need to add more configs for this release
> 
Got it, all sounds good.

> Cool. Let's go through this procedure tomorrow. We may automatize the release process (or document it!) via some scripts but it won't hurt if we do it manually first.

Thank you, really appreciate you reviewing this again with me.

> I now remember. Since the View Mode mapper uses "any" `type`:`Value` key found in the JSON, by having Video further down in the list we can catch any ADO that has a Video File (because as:video will contain files that have "type":"Video"). Same goes for Document, Image, etc. So we can keep both, Video Object and Movie Further up so they are selected first, and if not, as last resort, Video. Does that make sense?
> 

Ah ok, that makes sense. So: should that mapping be added back then? 

> I think I should take care of that one... Leave it for now

Ok, will leave this be.

> Google Group is more open I guess. Thanks!

Will post something about this tomorrow or a little later in the week--will give our wonderful EADC colleagues a heads up first. :)

Thanks again Diego!

DiegoPino commented:
@alliomeria all this is looking so good! Thanks a lot

alliomeria commented:
@DiegoPino, thank you again for your review and feedback about all of these! Thank you also for going over the json_encode_string process today. 

Following up on our discussions from earlier today, the latest commit here:
- adds the JSON API ready metadatadisplay_entity*.json files for all the templates found here (1,2,4,7,10,11, and now 12)
- also adds the corresponding updated versions of the multiplie_thumbnails_via_iiif_and_fontawesome and carousel_image templates

Please let me know if these need any additional corrections, or if the thumbnails & carousel templates should be swapped for different versions. Thank you!

DiegoPino reviewed (approved):
Super DUPER! Thanks @alliomeria 🥇
